src/mainnet/processors/legacy-staking.ts

```diff
@@ -0,0 +1,108 @@
+}
+
+export const setup = (processor: EvmBatchProcessor) => {
+  processor.addLog({
+    address: [LEGACY_OGN_STAKING],
+    topic0: [
+      legacyStakingAbi.events['Staked(address,uint256)'],
+      legacyStakingAbi.events['Staked(address,uint256,uint256,uint256)'],
+      legacyStakingAbi.events['Withdrawn(address,uint256)'],
+      legacyStakingAbi.events['Withdrawn(address,uint256,uint256)'],
+      legacyStakingAbi.events.StakesTransfered,
+    ].map((ev) => ev.topic),
+    range: { from },
+  })
+}
+
+export const process = async (ctx: Context) => {
+  const result: IProcessResult = {
+    legacyStakers: new Map<string, LegacyStaker>(),
+  }
+
+  for (const block of ctx.blocks) {
+    for (const log of block.logs) {
+      const firstTopic = log.topics[0]
+
+      if (![LEGACY_OGN_STAKING].includes(log.address.toLowerCase())) {
+        continue
+      }
+
+      if (firstTopic === legacyStakingAbi.events['Staked(address,uint256)'].topic) {
+        let { user, amount } = legacyStakingAbi.events['Staked(address,uint256)'].decode(log)
+        const staker = await _getStaker(ctx, user, result)
+        staker.inputAmount += amount
+        staker.balance += amount
+      } else if (firstTopic === legacyStakingAbi.events['Staked(address,uint256,uint256,uint256)'].topic) {
+        let { user, amount } = legacyStakingAbi.events['Staked(address,uint256,uint256,uint256)'].decode(log)
+        const staker = await _getStaker(ctx, user, result)
+        staker.inputAmount += amount
+        staker.balance += amount
+      } else if (firstTopic === legacyStakingAbi.events['Withdrawn(address,uint256)'].topic) {
+        let { user, amount } = legacyStakingAbi.events['Withdrawn(address,uint256)'].decode(log)
+        const staker = await _getStaker(ctx, user, result)
+        staker.outputAmount += amount
+        staker.balance -= amount
+      } else if (firstTopic === legacyStakingAbi.events['Withdrawn(address,uint256,uint256)'].topic) {
+        let { user, amount, stakedAmount } = legacyStakingAbi.events['Withdrawn(address,uint256,uint256)'].decode(log)
+        const staker = await _getStaker(ctx, user, result)
+        staker.outputAmount += stakedAmount
+        staker.rewardAmount += amount - stakedAmount
+        staker.balance -= stakedAmount
+      } else if (firstTopic === legacyStakingAbi.events.StakesTransfered.topic) {
+        let { toUser, fromUser } = legacyStakingAbi.events.StakesTransfered.decode(log)
+
+        const toStaker = await _getStaker(ctx, toUser, result)
+        const fromStaker = await _getStaker(ctx, fromUser, result)
+
+        toStaker.inputAmount += fromStaker.inputAmount
+        toStaker.outputAmount += fromStaker.outputAmount
+        toStaker.rewardAmount += fromStaker.rewardAmount
+        toStaker.balance += fromStaker.balance
+
+        fromStaker.inputAmount = 0n
+        fromStaker.outputAmount = 0n
+        fromStaker.rewardAmount = 0n
+        fromStaker.balance = 0n
```

Comment:
Why doesn't this use `numStakes` to determine what portion of the stake was transferred? Have you already determined that they're all complete transfers?